trackintel/preprocessing/positionfixes.py

```diff
@@ -386,3 +336,91 @@ def generate_triplegs(positionfixes, staypoints=None, *args, **kwargs):
     # todo: triplegs dataframe has use the index as id
     # todo: proposed fix: ret_triplegs = ret_triplegs.set_index('id')
     return ret_pfs, ret_tpls
+
+
+def _generate_staypoints_sliding_user(df, 
+                                      name_geocol, 
```

Comment:
Why do you need to have an explicit parameter for "name_geocol"? It is in `df` (which probably should be gdf) isn't it?

Reply:
I encountered this strange behaviour of not having `gdf.geometry` after `gdf.groupby()`, [here ](https://github.com/geopandas/geopandas/issues/1777)for open bugs.